MeshLib/convertMeshToGeo.cpp

```diff
@@ -133,12 +133,12 @@ bool convertMeshToGeo(const MeshLib::Mesh& mesh,
         addElementToSurface(*elements[i], id_map, *(*sfcs)[surfaceId]);
     }
 
-    std::for_each(sfcs->begin(), sfcs->end(), [](GeoLib::Surface* sfc) {
+    std::for_each(sfcs->begin(), sfcs->end(), [](GeoLib::Surface*& sfc) {
         if (sfc->getNumberOfTriangles() == 0)
         {
             delete sfc;
+            sfc = nullptr;
```

Comment:
This is tricky. *Technically* it shouldn't be possible to add surfaces without triangles. So most likely the test here is not necessary. However, if this *does* happen, then the surface shouldn't be removed here. It'll mess up the Data Explorer if there's suddenly a nullptr in a loaded geometry out of nowhere.

Reply:
Yes, that is true. But two lines below the vector is cleaned up. The problem here is: The surface containing zero triangles is deleted, but the pointer isn't changed. So the code below doesn't tidy up the vector. IMHO it is possible to remove the entire `for_each` construct and the following lines. What do you think?